build failed with vs2019 16.1.1 and windows sdk 10.0.18362

failed to cmake \tests\src\Interop\WinRT\NativeComponent, the output is 

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

BUILDTEST: Commencing build of native test components for x64/Debug
BUILDTEST: Using environment: "e:\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\\..\..\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.1.1
** Copyright (c) 2019 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86_x64'
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.
CMake Deprecation Warning at src/Interop/PInvoke/DllImportPath/CMakeLists.txt:11 (cmake_policy):
  The OLD behavior for policy CMP0037 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


Using Windows SDK version 10.0.18362.0 for WinRT interop tests.
CMake Error at src/Interop/WinRT/NativeComponent/CMakeLists.txt:42 (file):
  file FILE([TO_CMAKE_PATH|TO_NATIVE_PATH] path result) must be called with
  exactly three arguments.


-- Configuring incomplete, errors occurred!
See also "D:/git-repo/r/bin/tests/obj/Windows_NT.x64.Debug/Native/CMakeFiles/CMakeOutput.log".
BUILDTEST: Failed to generate test native component build project

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

--- a/tests/src/Interop/WinRT/NativeComponent/CMakeLists.txt
+++ b/tests/src/Interop/WinRT/NativeComponent/CMakeLists.txt
@@ -28,11 +28,8 @@ else()
   message("Using Windows SDK version ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION} for WinRT interop tests.")
 endif()
 
-set (PROGRAM_FILES_X86 "ProgramFiles(x86)")
-set (WINDOWS_KITS_DIR "$ENV{${PROGRAM_FILES_X86}}/Windows Kits/10")
-if (NOT EXISTS ${WINDOWS_KITS_DIR})
-  set (WINDOWS_KITS_DIR "$ENV{ProgramFiles}/Windows Kits/10")
-endif()
+get_filename_component(WINDOWS_KITS_DIR "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots;KitsRoot10]" ABSOLUTE CACHE)
+
 set (METADATA_DIR "${WINDOWS_KITS_DIR}/References/${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}/windows.foundation.foundationcontract/*" )
 set (REFERENCE_WINMDS
   "${WINDOWS_KITS_DIR}/References/${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}/Windows.Foundation.FoundationContract/*/Windows.Foundation.FoundationContract.winmd"

--- a/tests/src/Interop/WinRT/NativeComponent/CMakeLists.txt
+++ b/tests/src/Interop/WinRT/NativeComponent/CMakeLists.txt
@@ -28,7 +28,7 @@ else()
   message("Using Windows SDK version ${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION} for WinRT interop tests.")
 endif()
 
-get_filename_component(WINDOWS_KITS_DIR "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots;KitsRoot10]" ABSOLUTE CACHE)
+file(TO_CMAKE_PATH "$ENV{WindowsSdkDir}" WINDOWS_KITS_DIR)
 
 set (METADATA_DIR "${WINDOWS_KITS_DIR}/References/${CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION}/windows.foundation.foundationcontract/*" )
 set (REFERENCE_WINMDS

